crates/openvino/src/core.rs

```diff
@@ -51,17 +53,19 @@ impl Core {
     }
 
     /// Read model with model and weights loaded in memory.
-    pub fn read_model_from_buffer(
+    pub fn read_model_from_buffer<'a, T: Into<Option<&'a Tensor>>>(
         &mut self,
-        model_str: &str,
-        weights_buffer: &Tensor,
+        model_str: &[u8],
+        weights_buffer: T,
```

Comment:
I propose we move back to the `Option<&Tensor>` like you had; it's a bit more straightforward and avoids potential concerns with code size, etc.

Reply:
Sounds good 👍